### PR TITLE
fixes: app crash on selecting page switch event for query manager

### PR DESCRIPTION
**Pull request by arpitnath**
Fixes:
App crash for add page switch event for
- [x] queries (Query manager)
- [x] form widget
- [x] page event handler (page handler menu)

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-3338ay2 App is crashing when selecting switch page action from event manager](https://app.clickup.com/t/3338ay2)

File: frontend/src/Editor/QueryManager/QueryManager.jsx [hunk 1]
```diff
@@ -801,6 +807,7 @@ class QueryManagerComponent extends React.Component {
                     components={this.props.allComponents}
                     apps={this.props.apps}
                     popoverPlacement="top"
+                    pages={this.state.pages}
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/QueryManager.jsx:810]: @arpitnath Can we access it through `props` directly instead of setting `state`?

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@arpitnath Added a suggestion on `QueryManager`

**Comment by emidhun** [comment]
Verified the fixes now, and will have complete regression after all the fixes are done(v2).

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**